language/vm/src/file_format.rs

```diff
@@ -1710,6 +1717,7 @@ impl CompiledModule {
     /// into_module, i.e., script.into_module().into_script() == script.
     pub fn into_script(self) -> CompiledScript {
         let mut inner = self.into_inner();
+        precondition!(!inner.function_defs.is_empty());
```

Comment:
can you share what a `precondition` translates into?

Reply:
It compiles to nothing. Think of it as a comment that is also understood by MIRAI.